### DISCUSSION TITLE
Inquiry about energy scan and next releases

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#9813] [Q&A] [answered]
Hello,

I have 2 quick inquires:-
1- In scenarios where Openthread needs to initiate energy scan from radio platform while a transmission is currently in progress, what is the recommended response? Specifically, should the radio return an OT_ERROR_BUSY response under such circumstances?

2- is it planned for specific date of next openthread releases ? if yes then when ?

Thanks in advance

**Comment by jwhui** [comment, marked as answer]
> 1- In scenarios where Openthread needs to initiate energy scan from radio platform while a transmission is currently in progress, what is the recommended response? Specifically, should the radio return an OT_ERROR_BUSY response under such circumstances?

OpenThread will only call `otPlatRadioEnergyScan()` when the radio is in the receive or sleep states. Calls to `otPlatRadioEnergyScan()` are expected to succeed when in the receive or sleep states.

https://github.com/openthread/openthread/blob/c73829fdbf5b0019dc611e913674021492c3fa9e/src/core/mac/sub_mac.cpp#L793-L798

> 2- is it planned for specific date of next openthread releases ? if yes then when ?

OpenThread releases are tied to commits used by the Thread Test Harness. We will create a new tag when the Thread Test Harness moves forward to a later commit.

**Reply by testter2** [reply]
Thanks for detailed answer.
